wandb/integration/diffusers/resolvers/sdxl.py

```diff
@@ -0,0 +1,268 @@
+        pipeline: Any,
+        workflow_stage: str,
+        response: Response,
+        kwargs: Dict[str, Any],
+    ) -> Dict[str, Any]:
+        """Prepare the loggable dictionary for text-to-image workflow.
+
+        Arguments:
+            pipeline: (Any) The Diffusion Pipeline.
+            workflow_stage: (str) The name of the workflow stage.
+            response: (wandb.sdk.integration_utils.auto_logging.Response) The response from
+                the request.
+            kwargs: (Dict[str, Any]) Dictionary of keyword arguments.
+
+        """
+        prompt_logging = (
+            kwargs["prompt"]
+            if isinstance(kwargs["prompt"], list)
+            else [kwargs["prompt"]]
+        )
+        prompt2_logging = (
+            kwargs["prompt_2"]
+            if isinstance(kwargs["prompt_2"], list)
+            else [kwargs["prompt_2"]]
+        )
+        negative_prompt_logging = (
+            kwargs["negative_prompt"]
+            if isinstance(kwargs["negative_prompt"], list)
+            else [kwargs["negative_prompt"]]
+        )
+        negative_prompt2_logging = (
+            kwargs["negative_prompt_2"]
+            if isinstance(kwargs["negative_prompt_2"], list)
+            else [kwargs["negative_prompt_2"]]
+        )
+        images = (
+            decode_sdxl_t2i_latents(pipeline, response.images)
+            if kwargs["output_type"] == "latent"
+            else response.images
+        )
+        images = chunkify(images, len(prompt_logging))
+        for idx in range(len(prompt_logging)):
+            for image in images[idx]:
+                wandb.log(
+                    {
+                        f"Generated-Image/{workflow_stage}": wandb.Image(
+                            image,
+                            caption=f"Prompt-1: {prompt_logging[idx]}\nPrompt-2: {prompt2_logging[idx]}",
+                        )
+                    }
+                )
+                self.wandb_table.add_data(
+                    workflow_stage,
+                    prompt_logging[idx] if prompt_logging[idx] is not None else "",
+                    negative_prompt_logging[idx]
+                    if negative_prompt_logging[idx] is not None
+                    else "",
+                    prompt2_logging[idx] if prompt2_logging[idx] is not None else "",
+                    negative_prompt2_logging[idx]
+                    if negative_prompt2_logging[idx] is not None
+                    else "",
+                    wandb.Image(image),
+                )
+
+    def update_wandb_configs(
+        self, pipeline_configs: Dict[str, Any], kwargs: Dict[str, Any]
+    ) -> None:
+        """Update the configs of the W&B run.
+
+        Arguments:
+            pipeline_configs: (Dict[str, Any]) The configs corresponding to the
+                architecture of the pipeline.
+            kwargs: (Dict[str, Any]) Dictionary of keyword arguments.
+
+        """
+        if "workflow" not in wandb.config:
```

Comment:
are you guaranteed to have a run when you call this function?

Reply:
Yes, a run is guaranteed to be initialized before a resolver is called.